app/models/concerns/smooch_newsletter.rb

```diff
@@ -24,5 +24,14 @@ def toggle_subscription(uid, language, team_id, platform, workflow)
     def get_newsletter(team_id, language)
       TiplineNewsletter.where(team_id: team_id, language: language).last
     end
+
+    def unsubscribe_user_on_optout(json)
+      # unsubscribing user from all newsletters if the error code is 131050 and the platform is WhatsApp
+      if json.dig('destination', 'type') == 'whatsapp' && json.dig('error', 'underlyingError', 'errors', 0, 'code') == 131050
+        uid = json['appUser']['_id']
+        language = self.get_user_language(uid)
+        self.toggle_subscription(uid, language, self.config['team_id'].to_i, 'whatsapp', self.get_workflow(language))
```

Comment:
Sawy, since on WhatsApp there is no concept of opting-out for a language but only for the whole number, I suggest you unsubscribe the user from all newsletters.

Reply:
@caiosba the `toggle_subscription` method depend on language so you suggest to write a unsubscribe based on whole number only something like this `TiplineSubscription.where(uid: uid, team_id: team_id).destory_all` and in this case no need to call `self.toggle_subscription` method right?

Reply:
Yeah, you can implement a method `unsubscribe_from_all_language` in the Smooch Newsletter concern and call it here.

Reply:
That's a good point, Manu - I think that's fine because that message is already generic enough

Reply:
It says `You are currently not subscribed to our newsletter.`, I thought maybe it should make it clear that you unsubscribed from all newsletters, BUT I think that is probably rare enough that it could confuse most people?